### ISSUE TITLE
Backtrace on windows 7 32 bits image

### ISSUE THREAD

**Issue by ant1** [google/rekall#98]
Hi,

I have the following backtrace on a windows 7 32 bits image,  with both rekall 1.4.1 and rekall master from today:

```
% rekall -f win7x86.raw -v imageinfo
DEBUG:rekall.1:Logging level set to 10
DEBUG:rekall.1:Voting round with base: None
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.vmem.VMSSAddressSpace'> 
DEBUG:rekall.1:Failed instantiating VMSSAddressSpace: No base address space provided
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.macho.MACHOCoreDump'> 
DEBUG:rekall.1:Failed instantiating MACHOCoreDump: Must stack on another address space
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.vmem.VMemAddressSpace'> 
DEBUG:rekall.1:Failed instantiating VMemAddressSpace: No base address space provided
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.ewf.EWFAddressSpace'> 
DEBUG:rekall.1:Failed instantiating EWFAddressSpace: No base address space provided
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.crash.WindowsCrashBMP'> 
DEBUG:rekall.1:Failed instantiating WindowsCrashBMP: Must stack on another address space
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.elfcore.KCoreAddressSpace'> 
DEBUG:rekall.1:Failed instantiating KCoreAddressSpace: Must stack on another address space
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.crash.WindowsCrashDumpSpace32'> 
DEBUG:rekall.1:Failed instantiating WindowsCrashDumpSpace32: No base address space provided
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.crash.WindowsCrashDumpSpace64'> 
DEBUG:rekall.1:Failed instantiating WindowsCrashDumpSpace64: No base address space provided
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.elfcore.Elf64CoreDump'> 
DEBUG:rekall.1:Failed instantiating Elf64CoreDump: Must stack on another address space
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.lime.LimeAddressSpace'> 
DEBUG:rekall.1:Failed instantiating LimeAddressSpace: Must be layered on another address space.
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.aff4.AFF4AddressSpace'> 
DEBUG:rekall.1:Failed instantiating AFF4AddressSpace: Unable to open AFF4 volume
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.pmem.MacPmemAddressSpace'> 
DEBUG:rekall.1:Failed instantiating MacPmemAddressSpace: MacPmem would declare a YML device at /media/win7x86.raw_info
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.standard.FileAddressSpace'> 
DEBUG:rekall.1:Succeeded instantiating FileAddressSpace
DEBUG:rekall.1:Voting round with base: FileAddressSpace
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.vmem.VMSSAddressSpace'> 
DEBUG:rekall.1:Failed instantiating VMSSAddressSpace: Invalid VMware signature: 0xf000eef3
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.macho.MACHOCoreDump'> 
DEBUG:rekall.1:Failed instantiating MACHOCoreDump: Header signature invalid
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.vmem.VMemAddressSpace'> 
DEBUG:rekall.1:Failed instantiating VMemAddressSpace: Only VMEM files supported.
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.ewf.EWFAddressSpace'> 
DEBUG:rekall.1:Failed instantiating EWFAddressSpace: EWF signature not present
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.crash.WindowsCrashBMP'> 
DEBUG:rekall.1:Failed instantiating WindowsCrashBMP: Header signature invalid
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.elfcore.KCoreAddressSpace'> 
DEBUG:rekall.1:Failed instantiating KCoreAddressSpace: Header signature invalid
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.crash.WindowsCrashDumpSpace32'> 
DEBUG:rekall.1:Failed instantiating WindowsCrashDumpSpace32: Header signature invalid
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.crash.WindowsCrashDumpSpace64'> 
DEBUG:rekall.1:Failed instantiating WindowsCrashDumpSpace64: Header signature invalid
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.elfcore.Elf64CoreDump'> 
DEBUG:rekall.1:Failed instantiating Elf64CoreDump: Header signature invalid
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.lime.LimeAddressSpace'> 
DEBUG:rekall.1:Failed instantiating LimeAddressSpace: Invalid Lime header signature
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.aff4.AFF4AddressSpace'> 
DEBUG:rekall.1:Failed instantiating AFF4AddressSpace: Must stack on another address space
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.pmem.MacPmemAddressSpace'> 
DEBUG:rekall.1:Failed instantiating MacPmemAddressSpace: Must be mapped directly over a raw device.
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.standard.FileAddressSpace'> 
DEBUG:rekall.1:Failed instantiating FileAddressSpace: Must be first Address Space
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.mmap_address_space.MmapFileAddressSpace'> 
DEBUG:rekall.1:Failed instantiating MmapFileAddressSpace: Must be first Address Space
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.amd64.VTxPagedMemory'> 
DEBUG:rekall.1:Failed instantiating VTxPagedMemory: No more EPTs specified
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.standard.GlobalOffsetAddressSpace'> 
DEBUG:rekall.1:Failed instantiating GlobalOffsetAddressSpace: File offset not specified.
DEBUG:rekall.1:Trying <class 'rekall.plugins.addrspaces.pagefile.PagefilePhysicalAddressSpace'> 
DEBUG:rekall.1:Failed instantiating PagefilePhysicalAddressSpace: Pagefile not specified
INFO:rekall.1:Autodetected physical address space FileAddressSpace
DEBUG:rekall.1:Will detect profile using these Detectors: nt_index,osx,pe,windows_kernel_file,rsds,ntfs,linux
DEBUG:rekall.1:Opened url https://raw.githubusercontent.com/google/rekall-profiles/master/v1.0/inventory.gz
DEBUG:rekall.1:Opened local file /root/.rekall_cache/v1.0/inventory.gz
DEBUG:rekall.1:Opened url http://profiles.rekall-forensic.com/v1.0/inventory.gz
DEBUG:rekall.1:Opened local file /root/.rekall_cache/v1.0/inventory.gz
DEBUG:rekall.1:Opened local file /root/.rekall_cache/v1.0/nt/eprocess_index.gz
INFO:rekall.1:Loaded profile nt/eprocess_index from Local Cache Directory:/root/.rekall_cache (in 0.00921702384949 sec)
DEBUG:rekall.1:Opened local file /root/.rekall_cache/v1.0/nt/index.gz
INFO:rekall.1:Loaded profile nt/index from Local Cache Directory:/root/.rekall_cache (in 0.0124020576477 sec)
DEBUG:rekall.1:Opened local file /root/.rekall_cache/v1.0/OSX/index.gz
INFO:rekall.1:Loaded profile OSX/index from Local Cache Directory:/root/.rekall_cache (in 0.000488996505737 sec)
DEBUG:rekall.1:Opened local file /root/.rekall_cache/v1.0/pe.gz
INFO:rekall.1:Loaded profile pe from Local Cache Directory:/root/.rekall_cache (in 0.0234739780426 sec)
DEBUG:rekall.1:Opened local file /root/.rekall_cache/v1.0/ntfs.gz
INFO:rekall.1:Loaded profile ntfs from Local Cache Directory:/root/.rekall_cache (in 0.00160193443298 sec)
DEBUG:rekall.1:Skipped profile nt/GUID/FB5DCE282E8F4883BDB98B2BCB57BAB02 from https://raw.githubusercontent.com/google/rekall-profiles/master (Not in inventory)
DEBUG:rekall.1:Skipped profile nt/GUID/FB5DCE282E8F4883BDB98B2BCB57BAB02 from http://profiles.rekall-forensic.com (Not in inventory)
DEBUG:rekall.1:Will build local profile nt/GUID/FB5DCE282E8F4883BDB98B2BCB57BAB02
ERROR:rekall.1:Error: HTTP Error 404: Not Found
ERROR:rekall.1:Error: HTTP Error 404: Not Found
Extracting cabinet: /tmp/tmpG88Mzs/ntkrpamp.pd_
  extracting ntkrpamp.pdb

All done, no errors.
DEBUG:rekall.1:Opened local file /root/.rekall_cache/v1.0/mspdb.gz
INFO:rekall.1:Loaded profile mspdb from Local Cache Directory:/root/.rekall_cache (in 0.0130379199982 sec)
WARNING:rekall.1:Profile nt/GUID/FB5DCE282E8F4883BDB98B2BCB57BAB02 fetched and built. Please consider reporting this profile to the Rekall team so we may add it to the public profile repository.
DEBUG:rekall.1:Opened local file /root/.rekall_cache/v1.0/nt/GUID/FB5DCE282E8F4883BDB98B2BCB57BAB02.gz
INFO:rekall.1:Loaded profile nt/GUID/FB5DCE282E8F4883BDB98B2BCB57BAB02 from Local Cache Directory:/root/.rekall_cache (in 0.174033880234 sec)
DEBUG:rekall.1:No members specified for Struct _OBJECT_HEADER named _OBJECT_HEADER
WARNING:rekall.1:ExitTime has no offset in object _EPROCESS. Check that vtypes has a concrete definition for it.
WARNING:rekall.1:VadRoot has no offset in object _EPROCESS. Check that vtypes has a concrete definition for it.
WARNING:rekall.1:CreateTime has no offset in object _EPROCESS. Check that vtypes has a concrete definition for it.
WARNING:rekall.1:Token has no offset in object _EPROCESS. Check that vtypes has a concrete definition for it.
WARNING:rekall.1:Session has no offset in object _EPROCESS. Check that vtypes has a concrete definition for it.
WARNING:rekall.1:InheritedFromUniqueProcessId has no offset in object _EPROCESS. Check that vtypes has a concrete definition for it.
WARNING:rekall.1:UniqueProcessId has no offset in object _EPROCESS. Check that vtypes has a concrete definition for it.
WARNING:rekall.1:ImageFileName has no offset in object _EPROCESS. Check that vtypes has a concrete definition for it.
DEBUG:rekall.1:No members specified for Struct _EPROCESS named _EPROCESS
DEBUG:rekall.1:No members specified for Struct _EPROCESS named _EPROCESS
Traceback (most recent call last):
  File "/usr/local/bin/rekall", line 9, in <module>
    load_entry_point('rekall-core==0+unknown', 'console_scripts', 'rekall')()
  File "/usr/local/lib/python2.7/site-packages/rekall/rekal.py", line 104, in main
    user_session=user_session)
  File "/usr/local/lib/python2.7/site-packages/rekall/args.py", line 421, in parse_args
    "Plugin %s is not available for this configuration" % plugin_name)
rekall.plugin.PluginError: Plugin imageinfo is not available for this configuration
```

with volatility:
```
% vol.py -f win7x86.raw imageinfo
Volatility Foundation Volatility Framework 2.4
Determining profile based on KDBG search...

          Suggested Profile(s) : Win7SP0x86, Win7SP1x86
                     AS Layer1 : IA32PagedMemoryPae (Kernel AS)
                     AS Layer2 : FileAddressSpace (/home/antoine/win7x86.raw)
                      PAE type : PAE
                           DTB : 0x185000L
                          KDBG : 0x82f49c78
          Number of Processors : 4
     Image Type (Service Pack) : 1
                KPCR for CPU 0 : 0x82f4ad00
                KPCR for CPU 1 : 0x807c8000
                KPCR for CPU 2 : 0x8e700000
                KPCR for CPU 3 : 0x8e736000
             KUSER_SHARED_DATA : 0xffdf0000
           Image date and time : 2015-10-05 13:48:03 UTC+0000
     Image local date and time : 2015-10-05 15:48:03 +0200
```

**Comment by grrrrrrrrr** [comment]
Hey Antoine,

This is a problem with the pdb you downloaded from MS, it's missing structure information. Microsoft had some problems with pdb generation and they published some broken ones. We reported this issue and they fixed it but you still have the broken pdb in your cache.

In order to fix this, just remove any reference to FB5DCE282E8F4883BDB98B2BCB57BAB02 in your .rekall_cache directory (or just wipe the whole cache...).

I have already downloaded that pdb, confirmed that it now contains the missing structures, and uploaded the missing profile to the repository in https://github.com/google/rekall-profiles/commit/1e1f8e848ff75211941f9bc8bd7b572e0ac79f23 The next time you run (with a clean cache), that profile should be picked up automatically. Let me know if that fixes it!

Cheers,
-Andy
